tools/logicalimager/tsk_logical_imager.cpp

```diff
@@ -864,6 +864,15 @@ static bool hasTskLogicalImager(const TSK_TCHAR *image) {
             if (retval == 0 && fs_file != NULL && fs_file->meta != NULL) {
                 // found it
                 result = true;
+                TSK_FS_INFO *fsInfo = *fsListIter;
+                TSK_FS_TYPE_ENUM fileSystemType = fsInfo->ftype;
+                if (fileSystemType == TSK_FS_TYPE_FAT12 ||
+                    fileSystemType == TSK_FS_TYPE_FAT16 ||
+                    fileSystemType == TSK_FS_TYPE_FAT32 ||
+                    fileSystemType == TSK_FS_TYPE_FAT_DETECT) {
+                    consoleOutput(stderr, "Error: Writing to FAT device is not supported.");
+                    pressAnyKeyToExit(1);
```

Comment:
This I think should call handleExit. To avoid confusion, I think you should delete pressAnyKeyToExit entirely and fold it into handleExit.